### ISSUE TITLE
Mobile zoom. Support for pinch-zoom gesture on touch devices

### ISSUE THREAD

**Issue by DominikHoffendahl** [zcreativelabs/react-simple-maps#174]
**Problem**
The current implementation does not support the pinch-zoom gesture on mobile devices. Visit the example page for "Zooming and Panning" on your touch device: https://www.react-simple-maps.io/examples/zoom-pan/

**Expected behaviour**
It would be a nice addition to introduce the zooming functionality to touch devices. `d3.zoom` already provides the desired behaviour, although it moves the responsibility of DOM handling away from React and back to D3.

Checkout the [basic zoom example](https://www.d3-graph-gallery.com/graph/interactivity_zoom.html) of `d3.zoom` on your touch device.

- Panning used for movement
- Pinch to zoom
- DoubleTab / DoubleClick to zoom

Any ideas how to achieve the desired behaviour within the React context?

**Comment by daniel-karl** [comment]
Upvoting this heavily! 

Redditors are complaining they cannot use my COVID19 explorer https://daniel-karl.github.io/covid19-map/ nicely.

Today we need *this* component more than ever before 👍 

Some other things:
- [ ] Click on geography should zoom to its bounding box
- [ ] When loading the map it would be great to have good default zoom levels on any device (i.e. centered, zoomed to world bounding box)
- [ ] Alternative setting to zoom to users geo location based on their browser locale settings (or IP?)
- [ ] I have a hard time getting the tooltip example to work...I feel like I have everything in place (even memo), yet it doesn't show...

**Comment by DominikHoffendahl** [comment]
Hi @daniel-karl , thanks for upvoting my issue 👍

You should create separate issues for the bullets on your list, although they do not look like valid issues to me, as they ask for opinionated and custom functionality.
Let's not mixup things here!

Zoom into bounding box:
```
import { geoPath, geoMercator } from 'd3-geo';
...
// get the countries center coords
const center = projection.invert!(path.centroid(feature));
const [topLeftPoint, bottomRightPoint] = path.bounds(
    feature,
);

// get the right zoom level
const width = bottomRightPoint[0] - topLeftPoint[0];
const height = bottomRightPoint[1] - topLeftPoint[1];
const zoom = calculateMyOpinionatedZoom(width, height, mapSize);
```

Your second and third bullet is quite easy to achieve and should not be part of the framework.
Your last point should be a separate issue.

Getting back to the topic, I think that mobile support would provide a great leverage to this framework.

**Comment by daniel-karl** [comment]
@DominikHoffendahl Thanks a lot! I started a project to track COVID19, and I am relying on this map component here. I have more and more users who are having trouble with mobile, do you think you could have a look directly in my repo and improve whatever you can see I missed out on? I would totally appreciate your help with this! I have been working on the "Play" mode, so didn't even get to your suggestions above. Gotta catch some sleep now, 4 AM...
Code would be here: https://github.com/daniel-karl/covid19-map

**Comment by DominikHoffendahl** [comment]
Fixed with the latest release of version v1.0.0

**Comment by zimrick** [comment]
Note, it's v2 that fixes the mobile stuff using d3-zoom. Sorry it's a bit confusing since we did a very quick update from 1.0.0-beta to 1.0.0 and then v2, just to not cause any issues with semver. Hopefully from now on version updates will be a bit easier to follow.
